samples/java-spring-eventsourced-counter/src/it/java/com/example/CounterIntegrationWithRealPubSubTest.java

```diff
@@ -0,0 +1,91 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.time.Duration;
+import java.util.Base64;
+import java.util.concurrent.TimeUnit;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+// tag::class[]
+@SpringBootTest(classes = Main.class)
+@Import(TestkitConfig.class)
+@ActiveProfiles("with-pubsub")
+public class CounterIntegrationWithRealPubSubTest extends KalixIntegrationTestKitSupport { // <1>
+
+// end::class[]
+
+    private Duration timeout = Duration.of(10, SECONDS);
+
+    @Autowired
+    private WebClient webClient;
+
+    @Test
+    public void verifyCounterEventSourcedConsumesFromPubSub() {
+        WebClient pubsubClient = WebClient.builder()
+            .baseUrl("http://localhost:8085")
+            .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .build();
+
+        var counterId = "testRealPubSub";
+        var messageBody = buildMessageBody(
+            "{\"counterId\":\"" + counterId + "\",\"value\":20}",
+            CounterCommandFromTopicAction.IncreaseCounter.class.getName());
+
+        var projectId = "test";
+        var injectMsgResult = pubsubClient.post()
+            .uri("/v1/projects/{projectId}/topics/counter-commands:publish", projectId)
+            .bodyValue(messageBody)
+            .retrieve()
+            .toBodilessEntity().block();
+        assertTrue(injectMsgResult.getStatusCode().is2xxSuccessful());
+
+        await()
+            .ignoreExceptions()
+            .atMost(20, TimeUnit.SECONDS)
+            .until(() ->
+                    webClient.get()
+                        .uri("/counter/" + counterId)
+                        .retrieve()
+                        .bodyToMono(String.class)
+                        .block(timeout),
+                new IsEqual("\"20\"")
+            );
+    }
+    // end::test-topic[]
+
+    // builds a message in PubSub format, ready to be injected
+    private String buildMessageBody(String jsonMsg, String ceType) {
+        var data = Base64.getEncoder().encodeToString(jsonMsg.getBytes());
+        return "{\n" +
+            "    \"messages\": [\n" +
+            "        {\n" +
+            "            \"data\": \"" + data + "\",\n" +
+            "            \"attributes\": {\n" +
+            "                \"Content-Type\": \"application/json\",\n" +
+            "                \"ce-specversion\": \"1.0\",\n" +
+            "                \"ce-type\": \"" + ceType + "\"\n" +
+            "            }\n" +
+            "        }\n" +
+            "    ]\n" +
+            "}";
+    }
+
+// tag::class[]
+}
+// end::class[]
```

Comment:
This file is not used in documentation, I just figured it could be good to have a sample were we actually mix both tests using the test broker and a real instance. And since for this sample we actually already run the gpubsub on the pipeline, I thought it was good to add it here.